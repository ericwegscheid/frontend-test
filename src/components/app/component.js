import React, { Component, Fragment } from 'react'
import { MainView } from '../main-view'

export class AppComponent extends Component {
  constructor(props) {
    super(props)
  }

  render() {
    return <MainView />
  }
}
