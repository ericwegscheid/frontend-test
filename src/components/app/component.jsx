import React, { Component } from 'react'
import { MainView } from '../../components/main-view'
import './styles'

export class AppComponent extends Component {
  constructor(props) {
    super(props)
  }

  render() {
    return <MainView />
  }
}
