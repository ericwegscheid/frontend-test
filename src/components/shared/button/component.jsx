import React from 'react'
import './styles'

export const Button = props => {
  return (
    <button {...props}>
      { props.children }
    </button>
  )
}
