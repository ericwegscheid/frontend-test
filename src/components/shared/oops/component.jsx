import React from 'react'
import './styles'

export const Oops = props => {
  return (
    <div className="oops" {...props}>
      <div className="space-invader"></div>
      <h1>oops</h1>
      <h3>failure to contact the mothership</h3>
    </div>
  )
}
