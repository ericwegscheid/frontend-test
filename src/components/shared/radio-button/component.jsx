import { isFunction } from 'lodash'
import React, { Component } from 'react'
import './styles'

export class RadioButton extends Component {
  constructor(props) {
    super(props)

    this.state = {
      isSelected: props.isSelected,
      isDisabled: props.isDisabled,
    }
  }

  componentWillReceiveProps(nextProps) {
    const shouldUpdate =
      this.state.isSelected !== nextProps.isSelected ||
      this.state.isDisabled !== nextProps.isDisabled


    if (shouldUpdate) {
      this.setState({
        isSelected: nextProps.isSelected,
        isDisabled: nextProps.isDisabled,
      })
    }

    return shouldUpdate
  }

  onClick() {
    this.setState(state => ({
      isSelected: !state.isSelected,
    }))

    if (isFunction(this.props.onClick)) {
      this.props.onClick()
    }
  }

  render() {
    const { isSelected, isDisabled } = this.state

    return <div
      className={`control ${isDisabled ? 'disabled' : ''}`}
      onClick={this.onClick.bind(this)}
    >
      <div
        className={`radio-button ${isSelected ? 'selected' : ''}`}
      ></div>
      <label>{this.props.label}</label>
    </div>
  }
}
