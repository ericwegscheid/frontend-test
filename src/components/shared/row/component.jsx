import React from 'react'
import './styles'

export const Row = props => {
  return (
    <div className="row" {...props}>
      <div className="row-content">
        { props.children }
      </div>
    </div>
  )
}
