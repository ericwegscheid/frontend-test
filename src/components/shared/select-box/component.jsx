import { isFunction, map } from 'lodash'
import React, { Component } from 'react'
import './styles'

export class SelectBox extends Component {
  constructor(props) {
    super(props)

    this.state = {
      isOpen: false,
      options: {
        ...props.options,
      },
      selected: props.selected,
    }
  }

  componentWillReceiveProps(nextProps) {
    const shouldUpdate = this.state.selected !== nextProps.selected

    if (shouldUpdate) {
      this.setState({
        selected: nextProps.selected,
      })
    }

    return shouldUpdate
  }

  onClick() {
    this.setState(state => ({
      isOpen: !state.isOpen,
    }))
  }

  onSelectItem(selected) {
    this.setState(state => ({
      selected: selected,
    }))

    if (isFunction(this.props.onSelectItem)) {
      this.props.onSelectItem(selected)
    }
  }

  render() {
    const { isOpen, selected } = this.state

    return <div
      className={`control select-box ${isOpen ? 'open' : ''}`}
      onClick={this.onClick.bind(this)}
    >
      <ul>
        {
          map(this.props.options, v =>
            <li
              key={v.key}
              className={selected.key === v.key ? 'selected' : ''}
              onClick={this.onSelectItem.bind(this, v)}
            >
              <span>{v.value}</span>
            </li>
          )
        }
      </ul>
      <label>{this.props.label}<span>{selected.value}</span></label>
    </div>
  }
}
