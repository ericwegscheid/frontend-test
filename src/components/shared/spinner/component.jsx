import React from 'react'
import './styles'

export const Spinner = () => {
  return (
    <div className="spinner">
      <div className="mask"></div>
    </div>
  )
}
