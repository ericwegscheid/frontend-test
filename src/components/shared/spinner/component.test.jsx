import React from 'react'
import { Spinner } from './component.jsx'
import { shallow } from 'enzyme'

describe('Spinner component', () => {
  let component

  const renderComponent = props => {
    component = shallow(props)
  }

  it('should renderer snapshot', () => {
    renderComponent(
      <Spinner />
    )

    expect(component.html()).toMatchSnapshot()
  })
})
