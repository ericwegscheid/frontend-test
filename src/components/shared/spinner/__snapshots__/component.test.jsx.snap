// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component should renderer snapshot 1`] = `"<div class=\\"spinner\\"><div class=\\"mask\\"></div></div>"`;
